### PR TITLE
Kill Openvpn clients on session destroy

### PR DESCRIPTION
**Pull request by Waldz**
Fixes: #1838
Depends on: https://github.com/mysteriumnetwork/go-openvpn/pull/55

- Tracks in memory Openvpn's clients map - state of all currently connected clients
- Solved memory leaking of client map
- Allow same established session to be used by several clients as long as valid credentials are provided
- Kill all connected clients when session is destroyed by any error

### CODE REVIEW

File: services/openvpn/service/auth_handler_test.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2017 The "MysteriumNetwork/node" Authors.
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
```
Comment (anjmao) [inline services/openvpn/service/auth_handler_test.go:2]: How did you decided yo change this from 2017 to 2018? Was 2017 incorrect year?

File: services/openvpn/service/auth_handler.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
```
Comment (anjmao) [inline services/openvpn/service/auth_handler.go:2]: 2020

File: services/openvpn/service/auth_handler_test.go [hunk 3]
```diff
@@ -15,71 +15,74 @@
...
 )
 
-type fakeAuthenticatorStub struct {
-	username      string
-	password      string
-	called        bool
-	authenticated bool
-}
+func TestValidateReturnsFalseWhenNoSessionFound(t *testing.T) {
+	validator := mockAuthHandler(identity.Identity{}).validate
```
Comment (anjmao) [inline services/openvpn/service/auth_handler_test.go:42]: Why are you testing mock AuthHandler?

File: services/openvpn/service/manager.go [hunk 4]
```diff
@@ -299,10 +274,55 @@ func (m *Manager) ProvideConfig(_ string, sessionConfig json.RawMessage, conn *n
...
 		}
 	}
-	return &session.ConfigParams{SessionServiceConfig: vpnConfig, TraversalParams: traversalParams}, nil
+
+	destroy := func() {
+		log.Info().Msgf("Cleaning up session %s", sessionID)
+
+		sessionClients := m.openvpnClients.GetSessionClients(session.ID(sessionID))
+		for clientID := range sessionClients {
+			if err := m.openvpnAuth.ClientKill(clientID); err != nil {
```
Comment (anjmao) [inline services/openvpn/service/manager.go:283]: Code doesn't compile, there is no such method ClientKill. Are you sure this PR is ready?

Comment (Waldz) [reply services/openvpn/service/manager.go:283]: I will update dependency, after is merged https://github.com/mysteriumnetwork/go-openvpn/pull/55

File: services/openvpn/service/auth_handler_test.go (hunk 1 above)
Comment (Waldz) [reply services/openvpn/service/auth_handler_test.go:2]: by accident

File: services/openvpn/service/auth_handler.go (hunk 2 above)
Comment (Waldz) [reply services/openvpn/service/auth_handler.go:2]: It's old, i just moved it

File: services/openvpn/service/auth_handler_test.go (hunk 3 above)
Comment (Waldz) [reply services/openvpn/service/auth_handler_test.go:42]: It mocks several values to auth handler.
Renamed `mockAuthHandler` -> `createAuthHandler`

**Pull request merged**